src/helpers/script_helpers.sh

```diff
@@ -117,6 +145,15 @@ function run {
     preconditions "$@"
     steps > "$STEPS_FILE"
     run_steps "$STEPS_FILE" undoable
+
+    # Apply global previous-branch restoration
+    # only if we're at the git root.
+    # Any branch restoration logic for the other
+    # case needs to be handled in a custom way by
+    # the command itself
```

Comment:
Sorry it's been a while. At a quick glance, I really like this idea. I'll chew on it when I get back to finishing this feature.